Kumi.Game/Graphics/Backgrounds/BackgroundStack.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Kumi.Game.Graphics.Containers;
+using Kumi.Game.Screens.Backgrounds;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osuTK;
+
+namespace Kumi.Game.Graphics.Backgrounds;
+
+/// <summary>
+/// A stack of backgrounds that can be displayed by a <see cref="BackgroundScreen"/>, also there
+/// for handling transitions between backgrounds; as well as any effects that may be applied to
+/// them.
+/// </summary>
+public partial class BackgroundStack : CompositeDrawable
```

Comment:
We should probably use a ScreenStack instead, osu!framework will allow us to do that since it's just a `CompositeDrawable` behind the scenes and provides us with a better API to handle transitions between two background scenes.

Reply:
[There is one exists already.](https://github.com/irisu01/kumi/blob/background-screens/Kumi.Game/Screens/Backgrounds/BackgroundScreenStack.cs)

The sole reason this exists independently as well as part of `[BackgroundScreen](https://github.com/irisu01/kumi/blob/background-screens/Kumi.Game/Screens/Backgrounds/BackgroundScreen.cs)` is solely because of the fact theres an event to switch out backgrounds, and simply just swapping out background screen on a screen made to handle backgrounds will kindly just [push in the new screen altogether](https://github.com/ppy/osu-framework/wiki/Screens-and-Screen-Stacks#Only%20one%20screen%20can%20be%20active%20at%20a%20time%20within%20a%20single%20screen%20stack:~:text=Only%20one%20screen%20can%20be%20active%20at%20a%20time%20within%20a%20single%20screen%20stack.), which is an unwanted effect.

Reply:
Yes, you can stack `ScreenStack` but that isn't good enough.

Reply:
I suppose that's fine if we don't want a `Pop()` function, if I'm understanding this correctly?

Reply:
I doubt it. There should never be a case where there *isn't* a background.